### PR TITLE
Replace Schema Visitor with Schema Walker

### PR DESCRIPTION
**Pull request by ikitommi**
Re-implement Schema Visitor using Walker. Enables `find-first` to (almost) short-circuit when a match is found.

```clj
(m/find-first
  [:map
   [:x int?]
   [:y [:vector [:tuple
                 [:maybe int?]
                 [:or [:and {:salaisuus "turvassa"} boolean?] int?]
                 [:schema {:salaisuus "vaarassa"} false?]]]]
   [:z [:string {:salaisuus "piilossa"}]]]
  (fn [s _in _options]
    (some-> s m/properties :salaisuus)))
; => "turvassa"
```

### CODE REVIEW

File: src/malli/core.cljc [hunk 1]
```diff
@@ -935,6 +948,32 @@
...
+  ([?schema f options]
+   (-walk
+     (schema ?schema options)
+     (reify Walker
+       (-accept [_ s _ _] s)
+       (-inner [this s in options] (-walk s this in options))
+       (-outer [_ s c in options] (f s c in options)))
+     [] options)))
+
+(defn ^:no-doc find-first
```
Comment (rschmukler) [inline src/malli/core.cljc:963]: This seems like a useful / worth exporting into `clj-doc` - but maybe it should live under `malli.utils` like`merge`, `assoc`, etc?

File: README.md [hunk 2]
```diff
@@ -925,9 +925,9 @@ The identity visitor:
 Adding `:title` property to schemas:
 
 ```clj
-(m/accept
+(m/walk
   Address
-  (m/schema-visitor #(mu/update-properties % assoc :title (name (m/type %)))))
+  (m/schema-walker #(mu/update-properties % assoc :title (name (m/type %)))))
```
Comment (rschmukler) [inline README.md:930]: Could this be wrapped by `m/walk` - ie. If we are always going to have to wrap the function with `m/schema-walker`, why not just have `m/walk` do it for us?

Similar to how the `m/encode` will auto coerce non-reified schemas using `m/schema`

**Review by rschmukler — COMMENTED** [review]
A few random comments but overall this looks great!

**Review by rschmukler — APPROVED** [review]
A few random comments but overall this looks great!

File: README.md (hunk 2 above)
Comment (ikitommi) [reply README.md:930]: I think it covers the different cases nicely right now:

* `m/-walk` for any kind of walking, for library extensions
* `m/walk` with 4-arity fn for advanced postwalking (e.g. to map-syntax)
* `m/walk` + `m/schema-walker` to postwalk Schema instances to Schema instances, easy for many cases

File: src/malli/core.cljc (hunk 1 above)
Comment (ikitommi) [reply src/malli/core.cljc:963]: good point! Will move it to `malli.util` and make public & add to README.

Comment (ikitommi) [reply src/malli/core.cljc]: done

**Pull request merged**